cclib/parser/orcaparser.py

```diff
@@ -78,6 +78,20 @@ def after_parsing(self):
                     )
                     break
                 self.scfenergies[i] += dispersionenergy
+        
+        # ORCA prints singlet and triplet excited states separately, so the energies are out of order.
```

Comment:
AFAIK we don't make a promise that energies (states) are in complete energetic order. Should we?

Reply:
This is a fair point. In my mind I think it makes sense to re-order and it's not particularly difficult to do so. In particular, I think most programs most of the time will print excited states in increasing energy order, so to the user it will often seem like the order is being enforced. This is until they come across a strange case like this one where suddenly there's a drop in energy halfway through their excited states, which could easily look like a cclib bug (I assumed it was). Not a problem tho can remove if not?

Reply:
Nope, I agree. It exposes an implementation detail that singlets and triplets are computed separately and leaves the user to suffer when they tend to care more about the energy as you say.